src/Symfony/Component/Console/Command/InvokableCommand.php

```diff
@@ -0,0 +1,106 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Represents an invokable command.
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * @internal
+ */
+class InvokableCommand
+{
+    private readonly \ReflectionFunction $reflection;
+
+    public function __construct(
+        private readonly Command $command,
+        private readonly \Closure $code,
+    ) {
+        $this->reflection = new \ReflectionFunction($code);
+    }
+
+    /**
+     * Invokes a callable with parameters generated from the input interface.
+     */
+    public function __invoke(InputInterface $input, OutputInterface $output): int
+    {
+        $statusCode = ($this->code)(...$this->getParameters($input, $output));
+
+        if (null !== $statusCode && !\is_int($statusCode)) {
+            throw new LogicException(\sprintf('The command "%s" must return either void or an integer value in the "%s" method, but "%s" was returned.', $this->command->getName(), $this->reflection->getName(), get_debug_type($statusCode)));
+        }
+
+        return $statusCode ?? 0;
+    }
+
+    /**
+     * Configures the input definition from an invokable-defined function.
+     *
+     * Processes the parameters of the reflection function to extract and
+     * add arguments or options to the provided input definition.
+     */
+    public function configure(InputDefinition $definition): void
+    {
+        foreach ($this->reflection->getParameters() as $parameter) {
+            if ($argument = Argument::tryFrom($parameter)) {
+                $definition->addArgument($argument->toInputArgument());
+            } elseif ($option = Option::tryFrom($parameter)) {
+                $definition->addOption($option->toInputOption());
+            }
+        }
+    }
+
+    private function getParameters(InputInterface $input, OutputInterface $output): array
+    {
+        $parameters = [];
+        foreach ($this->reflection->getParameters() as $parameter) {
+            if ($argument = Argument::tryFrom($parameter)) {
+                $parameters[] = $argument->resolveValue($input);
+
+                continue;
+            }
+
+            if ($option = Option::tryFrom($parameter)) {
+                $parameters[] = $option->resolveValue($input);
+
+                continue;
+            }
+
+            $type = $parameter->getType();
+
+            if (!$type instanceof \ReflectionNamedType) {
+                continue;
+            }
+
+            $parameters[] = match ($type->getName()) {
+                InputInterface::class => $input,
+                OutputInterface::class => $output,
+                SymfonyStyle::class => new SymfonyStyle($input, $output),
+                Application::class => $this->command->getApplication(),
+                default => throw new RuntimeException(\sprintf('Unsupported type "%s" for parameter "$%s".', $type->getName(), $parameter->getName())),
+            };
+        }
+
+        return $parameters ?: [$input, $output];
```

Comment:
Once parameters without type are rejected, you can remove this default parameters inherited from the `execute` method.

```suggestion
        return $parameters;
```

Reply:
Hopefully this is an acceptable breaking change

Reply:
I don't think it's acceptable to have such a BC break in a minor version. https://github.com/symfony/symfony/pull/59340#discussion_r1904161097